pkg/apis/kops/kubeletconfig.go

```diff
@@ -56,5 +61,27 @@ func BuildKubeletConfigSpec(cluster *Cluster, instanceGroup *InstanceGroup) (*Ku
 		c.NodeLabels[k] = v
 	}
 
+	sv, err := util.ParseKubernetesVersion(cluster.Spec.KubernetesVersion)
+	if err != nil {
+		return c, fmt.Errorf("Failed to lookup kubernetes version: %v", err)
+	}
+
+	// --register-with-taints was available in the first 1.6.0 alpha, no need to rely on semver's pre/build ordering
+	sv.Pre = nil
+	sv.Build = nil
+	if sv.GTE(semver.Version{Major: 1, Minor: 6, Patch: 0, Pre: nil, Build: nil}) {
+		for i, t := range instanceGroup.Spec.Taints {
+			if c.Taints == nil {
+				c.Taints = make([]string, len(instanceGroup.Spec.Taints))
+			}
+			c.Taints[i] = t
+		}
+
+		// Enable scheduling since it can be controlled via taints.
+		// For pre-1.6.0 clusters, this is handled by taints.go
+		registerSchedulable := true
+		c.RegisterSchedulable = &registerSchedulable
+	}
```

Comment:
I think I'm missing some context around the potential conflict scenario.  c.Taints should always be empty on a pre-1.6 cluster, so it seems like we could safely error out in the 'else' case if anything at all is set in there. That's assuming the explicit master taints you mentioned are those set by tainter.go which, AFAIK, would never make their way into c.Taints.